### ISSUE TITLE
"ServerFaultCode: The operation is not supported on the object"

### ISSUE THREAD

**Issue by bradfitz** [vmware/govmomi#952]
I've used govc successfully in the past, and I have production jobs using it regularly, but today I'm trying to use govc with my homelab cluster, which it doesn't seem to like:

```
bard:~ $ govc ls
govc: ServerFaultCode: The operation is not supported on the object.
bard:~ $ govc about
govc: ServerFaultCode: The operation is not supported on the object.
```

Does that mean my VMUG EVALExperience license for my 3-node vSphere + vCenter HA setup doesn't include API access?

If so, can we get a better error message in govc telling users what the actual problem is?

Or do I need to enable the API server in vCenter somewhere? I've only ever worked with vSphere clusters others have created before. This home one is the first I've put together, and perhaps I screwed it up.

Sorry for the noob questions.

Thanks!

**Comment by dougm** [comment]
Hey Brad, I haven't used the vmug license before, I'll have to look into it.

Can you check if your license has the vimapi feature enabled:

 ```console
% govc license.ls -json -feature vimapi | jq .[].EditionKey
"esx.enterprisePlus.cpuPackage"
```

And share your edition keys if it doesn't (will help me track down this licence):
```console
% govc license.ls -json | jq .[].EditionKey
"eval"
"esx.enterprisePlus.cpuPackage"
"vc.standard.instance"
```

**Comment by dougm** [comment]
Then again, if `govc about` fails, you likely can't collect the license properties either.

**Comment by bradfitz** [comment]
Yeah,

```
bard:~ $ govc license.ls -json -feature vimapi
govc: ServerFaultCode: The operation is not supported on the object.

bard:~ $ govc license.ls -json
govc: ServerFaultCode: The operation is not supported on the object.
```

From the web UI, I see:

<img width="807" alt="screen shot 2017-12-11 at 12 03 25 pm" src="https://user-images.githubusercontent.com/2621/33851287-5bee13b0-de6b-11e7-9e88-9941022a0cb8.png">

Maybe I need to add more VMUG licenses or something.

But considering that I was able to set up an HA vCenter configuration, I'd be surprised if there was a license that allowed HA but did not also include API access.

**Comment by dougm** [comment]
Looking at this list: https://www.vmug.com/Join/EVALExperience

"VMware vSphere® ESXi Enterprise Plus with Operations Management™" should work if you assign it to your ESX hosts.

Normally the "eval" license that is included in the default install allows API access (until the eval expires).  I use this script to assign licenses: https://github.com/vmware/govmomi/blob/master/scripts/vcsa/README.md#licenses

**Comment by bradfitz** [comment]
I had a giant reply written here with all my debugging steps, but then as I was reading through the HTTP requests & responses from the `-debug` flag's output to `~/.govmomi/debug`, I kept setting `<userName>bradfitz</userName>` instead of my actual username `brad@vsphere.xxx.com`.

Whoops--- I had `GOVC_USER` set in my environment, not `GOVC_USERNAME`.

Sigh.

Sorry.

But a nicer error message would be nice, if such a thing is possible.

But the API server doesn't give you much to work with:

```
bard:~ $ cat /home/bradfitz/.govmomi/debug/2017-12-11T14-55-17.547742701/1-0002.req.headers ; echo
POST /sdk HTTP/1.1
Host: 10.0.0.150:443
Content-Type: text/xml; charset="utf-8"
Soapaction: urn:vim25/6.5
User-Agent: govc/0.16.1

bard:~ $ cat /home/bradfitz/.govmomi/debug/2017-12-11T14-55-17.547742701/1-0002.req.xml; echo
<?xml version="1.0" encoding="UTF-8"?>
<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/"><Header></Header><Body><AcquireLocalTicket xmlns="urn:vim25"><_this type="SessionManager">SessionManager</_this><userName>bradfitz</userName></AcquireLocalTicket></Body></Envelope>

bard:~ $ cat /home/bradfitz/.govmomi/debug/2017-12-11T14-55-17.547742701/1-0002.res.headers ; echo
HTTP/1.1 500 Internal Server Error
Content-Length: 566
Cache-Control: no-cache
Connection: Keep-Alive
Content-Type: text/xml; charset=utf-8
Date: Mon, 11 Dec 2017 22:55:17 GMT
X-Frame-Options: DENY

bard:~ $ cat /home/bradfitz/.govmomi/debug/2017-12-11T14-55-17.547742701/1-0002.res.xml ; echo
<?xml version="1.0" encoding="UTF-8"?>
<soapenv:Envelope xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/"
 xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
 xmlns:xsd="http://www.w3.org/2001/XMLSchema"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
<soapenv:Body>
<soapenv:Fault><faultcode>ServerFaultCode</faultcode><faultstring>The operation is not supported on the object.</faultstring><detail><NotSupportedFault xmlns="urn:vim25" xsi:type="NotSupported"></NotSupportedFault></detail></soapenv:Fault>
</soapenv:Body>
</soapenv:Envelope>
```

**Comment by dougm** [comment]
Argh, sorry Brad, this confusion is govc's fault.  Fixed in #955 so the error message in this case will be:
```
govc: ServerFaultCode: Cannot complete login due to an incorrect user name or password.
```
